packages/eslint-config-react/rules/base.js

```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  rules: {
+    'no-restricted-imports': [
+      'error',
+      {
+        paths: [
+          {
+            name: 'react',
+            importNames: ['FC'],
```

Comment:
Can we also include the possible use of `React.FC` that @a88zach mentioned in this comment – https://github.com/reside-eng/guidelines/pull/1#discussion_r751384284?

Reply:
Good idea – I added a rule from `typescript-eslint`. However, would that rule cause problems if a project was not using TypeScript?

Reply:
Hmm, is it possible to use `no-restricted-imports` like the this one? We shouldn't mix Typescript rules in here. This just further emphasizes our need to get our separate app configs that contain Typescript.

cc @prescottprue

Reply:
`no-restricted-imports` only works on import statements. Since TS namespaces are ambient, there is no import that can be used to trigger a lint error. As a result, we need to use a separate TS-only rule that is designed for preventing the use of specific types.

Reply:
Got it. Then let's leave it out for now until @prescottprue and team get the TS stuff baked into lint-config.

Reply:
Sounds good! Would you prefer I leave the rule in as a comment (with a Jira ticket reference to the TS lint config work), or should I remove it entirely?

Reply:
You can remove it. Instead, could you create the Jira ticket and just put that code in the ticket as a snippet?

Reply:
Will do!

Reply:
https://residenetwork.atlassian.net/browse/PLAT-1264